base/poco/NetSSL_OpenSSL/src/SecureSocketImpl.cpp

```diff
@@ -464,10 +478,20 @@ bool SecureSocketImpl::isLocalHost(const std::string& hostName)
 X509* SecureSocketImpl::peerCertificate() const
 {
 	std::lock_guard<std::recursive_mutex> lock(_mutex);
```

Comment:
All the methods are protected by our mutex.

I rally not sure that we need to have 2 mutex.